### ISSUE TITLE
Header stack lastIndex generates incorrect JSON for BMv2

### ISSUE THREAD

**Issue by ygottlieb-perspectalabs-com** [p4lang/p4c#1409]
Accessing a hs.lastIndex for a BMv2-based target generates a field expression:

```
"type": "field"
"value": [ "hs", "lastIndex"]
```

This is rejected at runtime by, e.g., simple_switch with an error:
```
Calling target program-options parser
Invalid reference to object of type 'header' with name 'test'
```

I believe that the correct code to generate would be something like
```
"type" : "expression",                              
"value" : {                                         
  "op": "last_stack_index",                       
  "left": null,                                   
  "right": {                                      
    "type": "header_stack",                     
    "value": "hs"                             
  }                                               
}                                                   
```

I've attached a sample P4 program that exhibits the behavior and two .json files, the first generates the error above, the second does not.

Thank you all for the hard work you put in to creating and maintaining these useful tools.

**Comment by ygottlieb-perspectalabs-com** [comment]
[lastIndex.txt](https://github.com/p4lang/p4c/files/2206013/lastIndex.txt)
[lastIndex.fixed.json.txt](https://github.com/p4lang/p4c/files/2206014/lastIndex.fixed.json.txt)
[lastIndex.json.txt](https://github.com/p4lang/p4c/files/2206015/lastIndex.json.txt)

**Comment by mbudiu-vmw** [comment]
This is a great bug report. Hopefully I have pushed a working fix in the PR already submitted for #1406.
